1-Authentication/7-sign-in-express-mfa/Api/index.js

```diff
@@ -0,0 +1,65 @@
+// Import necessary modules
+const express = require("express");
+const msal = require("@azure/msal-node");
+const app = express();
+const port = 3001;
+const { v4: uuidv4 } = require("uuid");
+const {
+  msalConfig,
+  GRAPH_ME_ENDPOINT,
+  TENANT_SUBDOMAIN,
+  TENANT_ID,
+} = require("./authConfig");
+
+var { fetch } = require("./fetch");
+
+// Middleware to parse incoming JSON requests
+app.use(express.json());
+
+const cca = new msal.ConfidentialClientApplication(msalConfig);
+
+// Function to acquire token using client credentials
+async function getAccessToken(tokenRequest) {
+  try {
+    const response = await cca.acquireTokenOnBehalfOf(tokenRequest);
+    return response.accessToken;
+  } catch (error) {
+    console.error("Error acquiring token:", error);
+    throw error;
+  }
+}
+
+// Define the updateUserInfo endpoint
+app.post("/updateUserInfo", async (req, res) => {
+  try {
+    const tokenRequest = {
+      oboAssertion: req.headers.authorization.replace("Bearer ", ""),
+      authority: `https://${TENANT_SUBDOMAIN}.ciamlogin.com/${TENANT_ID}`,
+      scopes: ["User.ReadWrite"],
+      correlationId: `${uuidv4()}`,
+    };
+
+    let accessToken = await getAccessToken(tokenRequest);
+    fetch(GRAPH_ME_ENDPOINT, accessToken, "PATCH", req.body)
+      .then((response) => {
+        if (response.status === 204) {
+          res.status(204);
+          res.json({ message: "Success" });
+        } else {
+          res.status(500);
```

Comment:
```suggestion
          res.status(502);
```

Gateway errors should be 502, however I would suggest also including the upstream error rather than hiding it. (i.e., include the result of `response.status` and the message's error) body